packages/x-data-grid/src/colDef/gridStringOperators.ts

```diff
@@ -38,6 +38,21 @@ export const getGridStringOperators = (
     },
     InputComponent: GridFilterInputValue,
   },
+  {
+    value: 'doesNotContain',
+    getApplyFilterFn: (filterItem: GridFilterItem) => {
+      if (!filterItem.value) {
+        return null;
+      }
+      const filterItemValue = disableTrim ? filterItem.value : filterItem.value.trim();
+
+      const filterRegex = new RegExp(escapeRegExp(filterItemValue), 'i');
+      return (value): boolean => {
+        return value != null ? !filterRegex.test(String(value)) : true;
+      };
```

Comment:
It would be great to extract a function that will handle both `contains` and `doesNotContain` operators since the only difference is the opposite return value.
Same for the other two negated operators.